core/src/main/java/org/apache/iceberg/rest/requests/PlanTableScanRequestParser.java

```diff
@@ -0,0 +1,124 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.List;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.ExpressionParser;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+public class PlanTableScanRequestParser {
+  private static final String SNAPSHOT_ID = "snapshot-id";
+  private static final String SELECT = "select";
+  private static final String FILTER = "filter";
+  private static final String CASE_SENSITIVE = "case-sensitive";
+  private static final String USE_SNAPSHOT_SCHEMA = "use-snapshot-schema";
+  private static final String START_SNAPSHOT_ID = "start-snapshot-id";
+  private static final String END_SNAPSHOT_ID = "end-snapshot-id";
+  private static final String STATS_FIELDS = "stats-fields";
+
+  private PlanTableScanRequestParser() {}
+
+  public static String toJson(PlanTableScanRequest request) {
+    return toJson(request, false);
+  }
+
+  public static String toJson(PlanTableScanRequest request, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(request, gen), pretty);
+  }
+
+  public static void toJson(PlanTableScanRequest request, JsonGenerator gen) throws IOException {
+    Preconditions.checkArgument(null != request, "Invalid request: planTableScanRequest null");
+    gen.writeStartObject();
+    if (request.snapshotId() != null) {
+      gen.writeNumberField(SNAPSHOT_ID, request.snapshotId());
+    }
+    if (request.select() != null && !request.select().isEmpty()) {
+      JsonUtil.writeStringArray(SELECT, request.select(), gen);
+    }
+    if (request.filter() != null) {
+      gen.writeStringField(FILTER, ExpressionParser.toJson(request.filter()));
+    }
+    if (request.caseSensitive() != null) {
+      gen.writeBooleanField(CASE_SENSITIVE, request.caseSensitive());
+    } else {
+      gen.writeBooleanField(CASE_SENSITIVE, true);
+    }
+    if (request.useSnapshotSchema() != null) {
+      gen.writeBooleanField(USE_SNAPSHOT_SCHEMA, request.useSnapshotSchema());
+    } else {
+      gen.writeBooleanField(USE_SNAPSHOT_SCHEMA, false);
+    }
```

Comment:
See my comment above, I think we should write the model in the library in a manner where request.useSnapshotSchema() always returns a concrete boolean value and can't be null. 

The spec says it's optional but that doesn't mean the reference implementation for the spec should enable that behavior for the in memory representation for these structures. The spec just gives that flexibility to clients and defines what the server should do in case these boolean fields are missing.

Reply:
Also newlines after if blocks please. I think this is a case where it's really helpful for readability.

Reply:
Will fix this thanks